### ISSUE TITLE
Dialog plug-in prompt text box issue with Japanese IME

### ISSUE THREAD

**Issue by sinbino** [quasarframework/quasar#5613]
**Describe the bug**
This is a problem with the dialog plug-in prompt text box.
When using the Japanese IME, it uses the enter key to confirm the phrase conversion.
This is an enter key to confirm the phrase conversion and the dialog should not be closed.
The quasar dialog plug-in prompt closes in this case.

**Codepen/jsFiddle/Codesandbox (required)**
https://codepen.io/pen/?&editable=true&editors=101

**To Reproduce**
Press the prompt of this codepen to input Japanese.
Since the dialog closes when the phrase is confirmed, you cannot enter multiple phrase.

**Expected behavior**
The dialog is not closed by pressing the Enter key in the Japanese IME phrase conversion mode.
This is always the correct behavior in standard form / input html.
Even when a form is submitted with the Enter key, the confirmation of the conversion mode of the Japanese IME is treated separately.

**Platform (please complete the following information):**
OS: Windows10, MacOS10.15
Node: v10.16.3
NPM: 6.9.0
Yarn: none
Browsers: Chrome78.0.3904.97, Firefox 70.0.1, safari 13.0.3, IE11.0.9600.18816
iOS: none
Android: none
Electron: none

**Comment by sinbino** [comment]
https://github.com/quasarframework/quasar/blob/dev/ui/src/components/dialog-plugin/DialogPlugin.js
I think that there is a cause in lines 121-126 of the above source.
With keyCode === 13 in keyup event, IME phrase confirmation and Press Enter Key in the text box cannot be distinguished.


https://garafu.blogspot.com/2015/09/javascript-ime-enter-event.html
According to the information on this site, there are some differences in the case of IME phrase confirmation.
* When IME phrase is confirmed: keypress event does not fire
* When IME phrase is confirmed: keycode of keydown event reports 229 instead of 13.


&lt;form @ submit = "onSubmit"&gt; &lt;input ...&gt; ... &lt;/ form&gt;
Or if you handle a submit event in the form above, it will never be fired with IME phrase confirmation.

**Comment by sinbino** [comment]
I found a description of keyCode about Input Method Editor on w3.org.
http://lists.w3.org/Archives/Public/www-dom/2010JulSep/att-0182/keyCode-spec.html

I verified the behavior when using IME in these environments:
* MacOS Catalina
  * Chrome78.0.3904.97
  * Firefox 70.0.1
  * safari 13.0.3
* Windows7
  * IE11.0.9600.18816

When the Enter key was pressed in IME conversion mode, the following behavior was commonly observed.
* Press Enter and the first keydown event reports keycode 229
* If you keep pressing Enter, the keydown event reports keyCode 13 for each repeat.
* The keyup event reports keyCode 13 when the key is released.

From this, I think it should detect the Enter key press with the following code:
```
let keyDownCode = null

function onKeyDown(event) {
  if(keyDownCode === null) {
    keyDownCode = event.keyCode
  }
}

function onKeyUp(event) {
  if(keyDownCode !== null) {
    if(keyDownCode == 13 && event.keyCode == 13) {
      // ENTER KEY IS PRESSED!!!
    }
    keyDownCode = null
  }
}
```

I think this problem occurs not only in Japanese but also in multibyte languages that use IME.

**Comment by pdanpdan** [comment]
Can you please check if in onKeyUp(event) event.isComposing is consistent? True on an enter key that selects a composition and false otherwise?

**Comment by sinbino** [comment]
In onKeyUp with enter key, event.isComposing returned false regardless of composition selection or not.

**Comment by pdanpdan** [comment]
Ok, I'm lost. Can you please tell me what keys I should press using the Japanese keyboard? And what kind of input mode I should use?

**Comment by sinbino** [comment]
The Japanese keyboard has a dedicated key for switching the input mode.
In the case of a US keyboard, after enabling Japanese IME on the OS, it seems necessary to switch the input mode using an alternative key.

For Windows:
* Switch input mode: [Alt]+[`]
* Select conversion candidates: [Space]
* Conversion decision: [Enter]

For MacOS:
* Switch input mode: [Ctrl]+[Space]
* Select conversion candidates: [Space]
* Conversion decision: [Enter]

If my explanation is not enough, please ask further questions.

**Comment by rstoenescu** [comment]
Fix will be available in "quasar" v1.4.5.
